types/opt/value.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package opt
+
+import (
+	"fmt"
+	"reflect"
+
+	jsonv2 "github.com/go-json-experiment/json"
+	"github.com/go-json-experiment/json/jsontext"
+)
+
+// Value is an optional value to be JSON-encoded.
+// With [encoding/json], a zero Value is marshaled as a JSON null.
+// With [github.com/go-json-experiment/json], a zero Value is omitted from the
+// JSON object if the Go struct field specified with omitzero.
```

Comment:
FYI, I commented on https://github.com/golang/go/issues/45669#issuecomment-2215356195 to promote `omitzero` back onto the proposal track for addition in v1 "json".